qoqo_qiskit/src/qoqo_qiskit/transpiler_helper/transpiler_helper.py

```diff
@@ -0,0 +1,165 @@
+
+from qoqo_qasm import QasmBackend, qasm_str_to_circuit
+from qoqo import Circuit, QuantumProgram
+from qoqo.measurements import (
+    PauliZProduct,
+    ClassicalRegister,
+    CheatedPauliZProduct,
+    Cheated,
+)
+from qiskit import QuantumCircuit, transpile
+from qiskit.qasm2 import dumps
+
+
+def transpile_with_qiskit(
+    circuits: Circuit | list[Circuit], transpilers: list[dict[str, str]]
+) -> Circuit:
+    """Use qiskit transpilers to transpile a qoqo circuit.
+
+    Args:
+        circuits (Circuit | List[Circuits]): qoqo circuit(s) to transpile.
+        transpilers (list[dict[str, str]]): transpilers to use.
+
+    Returns:
+        Circuit: transpiled qoqo circuit.
+    """
+    circuits_is_list = isinstance(circuits, list)
+    circuits = circuits if circuits_is_list else [circuits]
+
+    qasm_backend = QasmBackend(qasm_version="2.0")
+    qiskit_circuits = [
+        QuantumCircuit.from_qasm_str(qasm_backend.circuit_to_qasm_str(circuit))
+        for circuit in circuits
+    ]
+    for transpiler_args in transpilers:
+        qiskit_circuits = transpile(
+            qiskit_circuits,
+            backend=transpiler_args.get("backend"),
+            basis_gates=transpiler_args.get("basis_gates"),
+            inst_map=transpiler_args.get("inst_map"),
+            coupling_map=transpiler_args.get("coupling_map"),
+            backend_properties=transpiler_args.get("backend_properties"),
+            initial_layout=transpiler_args.get("initial_layout"),
+            layout_method=transpiler_args.get("layout_method"),
+            routing_method=transpiler_args.get("routing_method"),
+            translation_method=transpiler_args.get("translation_method"),
+            scheduling_method=transpiler_args.get("scheduling_method"),
+            instruction_durations=transpiler_args.get("instruction_durations"),
+            dt=transpiler_args.get("dt"),
+            approximation_degree=transpiler_args.get("approximation_degree") or 1.0,
+            timing_constraints=transpiler_args.get("timing_constraints"),
+            seed_transpiler=transpiler_args.get("seed_transpiler"),
+            optimization_level=transpiler_args.get("optimization_level"),
+            callback=transpiler_args.get("callback"),
+            output_name=transpiler_args.get("output_name"),
+            unitary_synthesis_method=transpiler_args.get("unitary_synthesis_method") or "default",
+            target=transpiler_args.get("target"),
+            hls_config=transpiler_args.get("hls_config"),
+            init_method=transpiler_args.get("init_method"),
+            optimization_method=transpiler_args.get("optimization_method"),
+            ignore_backend_supplied_default_methods=transpiler_args.get(
+                "ignore_backend_supplied_default_methods"
+            )
+            or False,
+            # num_processes=transpiler_args.get("num_processes"), in the documentation but not in the code.
```

Comment:
What does this mean exactly?

Reply:
It is in the qiskit documentation but since we use a qiskit version < 0.46 it's not implemented. I probably should just get rid of it